kafka/conn.py

```diff
@@ -120,11 +120,12 @@ def connect(self):
         if self.state is ConnectionStates.CONNECTING:
             # in non-blocking mode, use repeated calls to socket.connect_ex
             # to check connection status
+            ret = None
             request_timeout = self.config['request_timeout_ms'] / 1000.0
             try:
```

Comment:
strike that -- I believe your test case shows that an exception is raised on dns lookup failures:
```
gaierror: [Errno 8] nodename nor servname provided, or not known
```